### PR TITLE
Closes #1864: Add new breadcrumb-item-home class to front page segment.

### LINKED ISSUE #1864: Update default breadcrumb configuration for better user experience
## Motivation
We are working on improving our breadcrumb navigation to be more useful to site visitors.

### Is your feature request related to a problem? Please describe.
The current configuration for breadcrumbs can be confusing. Breadcrumbs only show up on third level pages or deeper. On third level pages there will only be one link present (the parent of the current page), and without any "trail", it isn't obvious that it is a breadcrumb menu and only looks like a floating word out of place.

### Describe the solution you'd like
Have breadcrumbs excluded from the home page. 
Have the breadcrumb trail always include the Home link. 
Include the current page (unclickable)

### Describe alternatives you've considered
Bob Liu from Libraries has additional UX feedback based on what they've done.

### Roles and Permissions considerations
A clear and concise description of how each of the following roles would be impacted by this change:
- Anonymous user: Should be able to view the breadcrumbs
- Content administrator: Should be able to override the breadcrumb settings.

### PR DESCRIPTION
**Pull request by joeparsons**
## Description
Adds new breadcrumb class to home item.

Currently contains changes from #1889 as well since changing Easy breadcrumb settings is broken without those changes.

## Related issues
Closes #1864 

## How to test
- Change Bootstrap version to latest dev
- Enable Easy breadcrumb "Include the current page as a segment in the breadcrumb" and "Include the front page as a segment in the breadcrumb" settings

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [x] Critical institutional link or brand change
- Minor release changes
   - [x] New feature
   - [x] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

### Drupal core
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

### Drupal contrib projects
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: themes/custom/az_barrio/az_barrio.theme [hunk 1]
```diff
@@ -539,12 +539,42 @@ function az_barrio_theme_suggestions_menu_alter(array &$suggestions, array $vari
...
+      }
+      else {
+        $add_home_class = TRUE;
+      }
+    }
+
+    foreach ($variables['breadcrumb'] as $key => &$item) {
+      $item['classes'] = 'breadcrumb-item';
+      if ($add_home_class && $key === 0) {
+        $item['classes'] .= ' breadcrumb-item-home';
```
Comment (trackleft) [inline themes/custom/az_barrio/az_barrio.theme:562]: This needs to be a string?

Comment (joeparsons) [reply themes/custom/az_barrio/az_barrio.theme:562]: The breadcrumb template provided by core doesn't have an existing classes or attributes variable.

File: themes/custom/az_barrio/az_barrio.theme [hunk 2]
```diff
@@ -539,12 +539,42 @@ function az_barrio_theme_suggestions_menu_alter(array &$suggestions, array $vari
...
- *  Implements template_preprocess_block().
+ * Implements hook_preprocess_HOOK() for breadcrumb templates.
+ */
+function az_barrio_preprocess_breadcrumb(&$variables) {
+  if (\Drupal::moduleHandler()->moduleExists('easy_breadcrumb')) {
+    $add_home_class = FALSE;
+    $config = \Drupal::config('easy_breadcrumb.settings');
+    if ($config->get('include_home_segment')) {
+      if (\Drupal::service('path.matcher')->isFrontPage()) {
+        if ($config->get('hide_single_home_item')) {
```
Comment (joeparsons) [inline themes/custom/az_barrio/az_barrio.theme:550]:
```suggestion
        if (!$config->get('hide_single_home_item')) {
```

**Comment by danahertzberg** [comment]
- [ ] Remove homepage check, only apply class to first item if include home segment option is checked
- [ ] Update breadcrumb block configuration to hide on front page
- [ ] Update default settings to match screenshot

screencapture-7b5e2985-cedf-4a13-ab91-3a35916c835d-pr-1899-probo-build-admin-config-user-interface-easy-breadcrumb-2022-09-21-14_07_34

File: themes/custom/az_barrio/az_barrio.theme [hunk 3]
```diff
@@ -539,12 +539,42 @@ function az_barrio_theme_suggestions_menu_alter(array &$suggestions, array $vari
...
+          $add_home_class = TRUE;
+        }
+      }
+      else {
+        $add_home_class = TRUE;
+      }
```
Comment (joeparsons) [inline themes/custom/az_barrio/az_barrio.theme:556]:
```suggestion
      $add_home_class = TRUE;
```

**Comment by danahertzberg** [comment]
Thanks for the updates @joeparsons! The functionality is looking good. One thing I noticed is that the breadcrumbs follow the URL alias rather than the menu structure.

For example, https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-1899.probo.build/pages/annual-events displays breadcrumbs that follow the custom URL alias rather than the menu structure.

Is this expected? If so, I think it's easily communicated.

**Comment by trackleft** [comment]
FYI I found this page https://www.drupal.org/docs/8/modules/easy-breadcrumb/configuration-options-for-easy-breadcrumb
Possibly useful.

**Comment by joeparsons** [comment]
> Is this expected? If so, I think it's easily communicated.

@danahertzberg, yes, I think that the breadcrumb trail provided by Drupal core (and modified by the Easy Breadcrumb module) are based on URL aliases, not menu links.

This is from the Easy Breadcrumb module project page:
>Breadcrumbs use the current URL (path alias) and the current page title to build the crumbs.

**Comment by danahertzberg** [comment]
Thanks so much for clarifying! As long as it's expected, I'm good. Approving now!

**Review by danahertzberg — APPROVED** [review]

**Comment by trackleft** [comment]
Probably better that it follows the path since our menu structure has nolinks at the top level

**Review by bberndt-uaz — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Pull request merged**